### PR TITLE
Set include and link dirs explicitly for macOS

### PR DESCRIPTION
**Pull request by oschwald**

### CODE REVIEW

File: .github/workflows/test-libmaxminddb.yml [hunk 1]
```diff
@@ -45,13 +45,35 @@ jobs:
...
-      - name: Build with Werror and Wall
+      - name: Build with Werror and Wall (not macOS)
         run: python setup.py build
+        if: matrix.platform != 'macos-latest'
         env:
           CFLAGS: "-Werror -Wall -Wextra"
 
       - name: Test with tox (system libmaxminddb)
         run: tox
+        if: matrix.platform != 'macos-latest'
```
Comment (ugexe) [inline .github/workflows/test-libmaxminddb.yml:59]: I don't know if its necessarily a better strategy, but one pattern I've used for platform specific github action workarounds is to set environment variables for specific platforms. Something like:
```
- name: "Work around macos arm64 homebrew directory changes"
  if: runner.os == 'macos' and runner.arch = 'arm64'
  run: |
    echo "CFLAGS=-I/opt/homebrew/include" >> $env:GITHUB_ENV
    echo "LDFLAGS=-L/opt/homebrew/lib" >> $env:GITHUB_ENV

- name: Build with Werror and Wall
    run: python setup.py build
    env:
      CFLAGS: "${{ env.CFLAGS }} -Werror -Wall -Wextra"

- name: Test with tox (system libmaxminddb)
    run: tox
```

Comment (oschwald) [reply .github/workflows/test-libmaxminddb.yml:59]: That does seem a bit cleaner. I can try doing that.

File: .github/workflows/test-libmaxminddb.yml [hunk 2]
```diff
@@ -44,36 +44,19 @@ jobs:
         run: sudo apt install libmaxminddb-dev
         if: matrix.platform == 'ubuntu-latest'
 
+      - name: "Work around macos arm64 homebrew directory changes"
+        if: runner.os == 'macos' and runner.arch = 'arm64'
```
Comment (ugexe) [inline .github/workflows/test-libmaxminddb.yml:48]: This might have to be `macOS`

Comment (oschwald) [reply .github/workflows/test-libmaxminddb.yml:48]: It seems like it worked. :shrug:

Comment (ugexe) [inline .github/workflows/test-libmaxminddb.yml:48]: Also `runner.arch = ` -> `runner.arch ==`

Comment (ugexe) [reply .github/workflows/test-libmaxminddb.yml:48]: Yeah I'm confused. It should be
```
if: runner.os == 'macOS' && runner.arch == 'ARM64'
```
See Annotations from https://github.com/maxmind/MaxMind-DB-Reader-python/actions/runs/8897863643
```
[Invalid workflow file: .github/workflows/test-libmaxminddb.yml#L48](https://github.com/maxmind/MaxMind-DB-Reader-python/actions/runs/8897863643/workflow)
The workflow is not valid. .github/workflows/test-libmaxminddb.yml (Line: 48, Col: 13): Unexpected symbol: 'and'. Located at position 22 within expression: runner.os == 'macos' and runner.arch = 'arm64'
```

Comment (oschwald) [reply .github/workflows/test-libmaxminddb.yml:48]: Oh, I would have expected some indication that failed in the build statuses or via email. I'll fix that up.

**Comment by ugexe** [comment]
Hmmm. I found https://github.com/actions/runner-images/discussions/9266#discussioncomment-8406444 which mentions:

> The problem is that while this maps on X64 for some reason the runner images use the lowercase arm64 rather than following the convention with ARM64. Perhaps this is the issue that needs fixing instead?

So maybe it does need to be `arm64`, unlike what is mentioned on https://docs.github.com/en/actions/learn-github-actions/variables 😞

**Comment by oschwald** [comment]
> So maybe it does need to be arm64

I am not sure that is the issue. That seems to be running. I thought it might be `$env:GITHUB_ENV` vs `"$GITHUB_ENV"`, which is mentioned [here](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-of-writing-an-environment-variable-to-github_env), but that didn't seem to help.

File: .github/workflows/test-libmaxminddb.yml [hunk 3]
```diff
@@ -48,10 +48,16 @@ jobs:
...
         if: matrix.platform == 'macos-latest'
 
-      - name: Build with Werror and Wall
+      - name: "Work around macos arm64 homebrew directory changes"
+        if: runner.os == 'macOS' && runner.arch == 'ARM64'
+        run: |
+          echo "CFLAGS=-I/opt/homebrew/include" >> "$GITHUB_ENV"
+          echo "LDFLAGS=-L/opt/homebrew/lib" >> "$GITHUB_ENV"
+
+      - name: Build with Werror and Wall (not macOS)
```
Comment (ugexe) [inline .github/workflows/test-libmaxminddb.yml:57]: The `(not macOS)` can be removed

Comment (oschwald) [reply .github/workflows/test-libmaxminddb.yml]: Thanks! I force-pushed it as a single commit to reduce the commit noise.

**Review by ugexe — APPROVED** [review]

**Pull request merged**